### DISCUSSION TITLE
PDF margins crop

### DISCUSSION THREAD

**Discussion by pajakpawel** [pymupdf/PyMuPDF#1023] [Looking for help] [answered]
Hey!
I need to crop margins (left, right, bottom and top) from every page of a given PDF. 
I found Issue #617 where I read that cropping pages margins is pretty easy when I know margins values.
And here comes my question. How can I get margins values for current page?

**Comment by JorjMcKie** [comment, marked as answer]
That's not easy, because there is no official PDF property to read this out.
You only can compute the maximum (union) rectangle for things on the page.
This computes the union of all text and images:
```python
rect = fitz.Rect()  # start empty
for b in page.get_text("blocks"):
    rect |= b[:4]  # join block rectangle into the total
```
If you want to be overly diligent, also look for any drawings (these are **_not_** images!):
```python
for path in page.get_drawings():
    rect |= path["rect"]
```

**Reply by pajakpawel** [reply]
That works great, thanks a lot!
